src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -72,15 +72,15 @@ public class OpenApiDocument : IOpenApiSerializable, IOpenApiExtensible
         /// </summary>
         public OpenApiDocument(OpenApiDocument document)
         {
-            Workspace = new(document.Workspace);
-            Info = new(document.Info);
-            Servers = new List<OpenApiServer>(document.Servers);
-            Paths = new(document.Paths);
-            Components = new(document.Components);
-            SecurityRequirements = new List<OpenApiSecurityRequirement>(document.SecurityRequirements);
-            Tags = new List<OpenApiTag>(document.Tags);
-            ExternalDocs = new(document.ExternalDocs);
-            Extensions = new Dictionary<string, IOpenApiExtension>(document.Extensions);
+            Workspace = new(document?.Workspace);
```

Comment:
```suggestion
            Workspace = document?.Workspace == null ? null : new(document?.Workspace);
```
what if the workspace was null on the passed document? now you have a value when you shouldn't.

(and repeat for object values)

Reply:
Okay yeah that makes sense. Resolving..